### PR TITLE
Add "Invalid Tag External Documentation URL" query for OpenAPI Closes#2882

### LINKED ISSUE #2882: Add "Invalid Tag External Documentation URL" query for OpenAPI
### Platform
OpenAPI

### Description
Tag External Documentation URL should be a valid URL

### PR DESCRIPTION
**Pull request by rafaela-soares**
Signed-off-by: Rafaela Soares rafaela.soares@checkmarx.com

Closes #2882

**Proposed Changes**
- Added "Invalid Tag External Documentation URL" query for OpenAPI (it checks if tags.externalDocs.url has aN invalid URL)

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/openAPI/invalid_tag_external_documentation_url/metadata.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": "5aea1d7e-b834-4749-b143-2c7ec3bd5922",
+  "queryName": "Invalid Tag External Documentation URL",
+  "severity": "INFO",
+  "category": "Best Practices",
+  "descriptionText": "Tag External Documentation URL should be a valid URL",
+  "descriptionUrl": "https://swagger.io/specification/#tag-object",
```
Comment (joaoReigota1) [inline assets/queries/openAPI/invalid_tag_external_documentation_url/metadata.json:7]: I believe this description url is more accurate: https://swagger.io/specification/#external-documentation-object

File: assets/queries/openAPI/invalid_tag_external_documentation_url/query.rego [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	not openapi_lib.is_valid_url(url)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("tags.name=%s", [name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("tags[%d].externalDocs.url has a valid URL", [x]),
+		"keyActualValue": sprintf("tags[%d].externalDocs.url has an invalid URL", [x]),
+	}
+}
```
Comment (joaoReigota1) [inline assets/queries/openAPI/invalid_tag_external_documentation_url/query.rego:20]: Tags can also be a field of  a Operation object: https://swagger.io/specification/#operation-object
This query is only checking tags inside OpenAPI Object.

**Review by joaoReigota1 — CHANGES_REQUESTED** [review]

File: assets/queries/openAPI/invalid_tag_external_documentation_url/query.rego (hunk 2 above)
Comment (rafaela-soares) [reply assets/queries/openAPI/invalid_tag_external_documentation_url/query.rego:20]: In the Operation object, the field 'tags' is a list of strings. There is no access to 'externalDocs'

**Review by felipe-avelar — DISMISSED** [review]
LGTM

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**